src/mithril.postgrest.js

```diff
@@ -8,30 +8,42 @@
     }
 }(function (m, _, localStorage) {
   var postgrest = {};
-  postgrest.init = function(apiPrefix){
 
-    postgrest.authenticate = function(options){
-      var deferred = m.deferred(),
-          token = localStorage.getItem("postgrest.token");
-      if(_.isString(token) && token.length > 0){
-        deferred.resolve({token: token});
-      }else{
-        m.request(options).then(function(data){
-          localStorage.setItem("postgrest.token", data.token);
-          deferred.resolve({token: data.token});
-        });  
-      }
-      postgrest.request = function(options){
-        return m.request(_.extend(options, {url: apiPrefix + options.url, config: function(){} }));
-      };
+  var xhrConfig = function(xhr){
+    if(token()){
+      xhr.setRequestHeader("Authorization", "Bearer " + token());
+    }
+    return xhr;
+  };
+
+  var token = function(){
```

Comment:
What about making this function a getter/setter? This way it would centralize all localStorage manipulation in one point.